File: redmond/models/src/main/java/ar/edu/itba/bd2/redmond/model/enums/TransactionStatus.java
package ar.edu.itba.bd2.redmond.model.enums;

public enum TransactionStatus {
    PENDING,
    APPROVED,
    REJECTED;

    public static TransactionStatus fromString(String status) {
        for(TransactionStatus transactionStatus : TransactionStatus.values()) {
            if(transactionStatus.name().equalsIgnoreCase(status)) {
                return transactionStatus;
            }
        }

        throw new IllegalArgumentException("Invalid transaction status: " + status);
    }
}

File: bank/src/main/kotlin/com/redmond/bankapi/models/TransactionStatus.kt
package com.redmond.bankapi.models

import com.redmond.bankapi.exceptions.InvalidStatusException

enum class TransactionStatus {
    PENDING,
    APPROVED,
    REJECTED;

    companion object {
        fun fromString(value: String): TransactionStatus {
            return when(value.lowercase()) {
                "pending" -> PENDING
                "approved" -> APPROVED
                "rejected" -> REJECTED
                else -> throw InvalidStatusException("Invalid status: '$value'")
            }
        }
    }
}